packages/app/src/systems/Transaction/utils/error.tsx

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+export type VMApiError = {
+  request: any;
+  response: {
+    errors: {
+      locations: any;
+      message: string;
+      path: string[];
+    }[];
+  };
+};
+
+export type VmErrorType = 'InsufficientInputAmount';
+export type InsufficientInputAmountError = {
+  asset: string;
+  expected: string;
+  provided: string;
+};
+
+export type GroupedError = {
+  [key: string]: any;
+  InsufficientInputAmount?: InsufficientInputAmountError[];
+};
+
+// typeguard
+const isVmErrorTypeKey = <K extends string & keyof GroupedError>(
+  properties: K[],
+  vmError: string
+): vmError is K => {
+  return (properties as string[]).includes(vmError);
+};
+
+export const getGroupedErrors = (rawErrors?: { message: string }[]) => {
```

Comment:
Why we need this?

Reply:
this file has some error helpers.

in `getGroupedErrors` it will return like:
![image](https://user-images.githubusercontent.com/8636507/198901745-d6c65c08-b637-477f-91fe-b69cfc64afe1.png)

then in the screens, we have the ability to add special treatments to any errors we want.
for example:
```
const { groupedErrors } = useTxApprove();

if (groupedErrors?.InsufficientInputAmount) {
  // do something specific to this error
}
```